pallets/cash/src/events.rs

```diff
@@ -70,9 +70,12 @@ fn fetch_eth_blocks(
 ) -> Result<ChainBlocks, EventError> {
     debug!("Fetching Eth Blocks [{}-{}]", from, to);
     let mut acc: Vec<<Ethereum as Chain>::Block> = vec![];
-    for block_number in from..to {
+    for block_number in from..=to {
         match fetch_eth_block(block_number, eth_starport_address) {
             Ok(block) => {
+                // if let Some(last) = acc.last() {
+                //     assert!(last.hash == block.parent_hash, "Fetched blocks do not form chain");
```

Comment:
worth it or not?

Reply:
On the worker side? Good ❓

Reply:
I think the consequences of this check need to be considered in the wider context of how reorgs work. I'm not super familiar with it. Aside from what the knock-on effects are, `assert` would not necessarily be the thing to do, should be `return Err(...)`